bundles/org.openhab.binding.konnected/src/main/java/org/openhab/binding/konnected/internal/KonnectedHandlerFactory.java

```diff
@@ -61,26 +66,25 @@ protected void activate(ComponentContext componentContext) {
         super.activate(componentContext);
         Dictionary<String, Object> properties = componentContext.getProperties();
         callbackUrl = (String) properties.get("callbackUrl");
-        KonnectedHTTPServlet servlet = registerWebHookServlet();
-        this.servlet = servlet;
+        try {
+            this.servlet = registerWebHookServlet();
+        } catch (KonnectedWebHookFail e) {
+            logger.error("Failed registering Konnected servlet - binding is not functional!", e);
+        }
     }
 
     @Override
     protected void deactivate(ComponentContext componentContext) {
         super.deactivate(componentContext);
-        servlet.deactivate();
+        httpService.unregister(alias);
     }
 
     @Override
     protected @Nullable ThingHandler createHandler(Thing thing) {
         KonnectedHandler thingHandler = new KonnectedHandler(thing, '/' + BINDING_ID, createCallbackUrl(),
                 createCallbackPort());
         logger.debug("Adding thinghandler for thing {} to webhook.", thing.getUID().getId());
-        try {
-            servlet.add(thingHandler);
-        } catch (KonnectedWebHookFail e) {
-            logger.trace("there was an error adding the thing handler to the servlet: {}", e.getMessage());
-        }
+        servlet.add(thingHandler);
```

Comment:
If the registration of the servlet will fail during activation of the factory we will face a NPE here. Should we handle that?

Reply:
Valid point. I've added a null check.